src/bfs_tree.c

```diff
@@ -77,106 +174,122 @@ BFSNode *bfs_tree_create_node(char fname[BFS_MAX_NAME_LENGTH]) {
+    if(check_pointer(root)) {
+        return EMPTY_NODE;
+    }
+    else if(strcmp(root->fname, s) == 0) {
+        tmp = root;
     }
     else {
-        BFSNode *walker = root->child;
+        bfs_tree_get_child(root->child, s);
+        bfs_tree_get_child(root->next, s);
+    }
+    return tmp;
+}
 
-        while(walker->next != NULL &&
-              bfs_strsplit(path, "/", i) != walker->fname) {
-            walker = walker->next;
-        }
+void bfs_tree_print(BFSTree root, char *arg) {
+    if(root == NULL) {
+        return;
+    }
+    else if(root->child == NULL && root->fpermissions == BFS_NODE_IS_DIR) {
+        bfs_pferror(stderr, "Empty directory", __LINE__, __FILE__,
+                    BFS_WARNING);
+        newline;
+        return;
+    }
 
-        while(i < bfs_count_delim(path, '/')) {
-            if(bfs_strsplit(path, '/', i) != walker->fname) {
-                // The directory doesn't exist, so need to be created first
-                BFSNode *node = bfs_tree_create_node(bfs_strsplit(path, "/",
-                                                                   i));
-                walker->next = node;
-                node->father = walker->father;
+    root = root->child;
+    
+    if(arg == NULL) {
+        newline;
+        while(root != NULL) {
+            if(root->fpermissions == BFS_NODE_IS_DIR) {
+                printf("%s%s%s    ", ANSI_COLOR_BLUE, root->fname,
+                    ANSI_COLOR_RESET);
             }
             else {
-                if(walker->child == NULL) {
-                    BFSNode *node = bfs_tree_create_node(
-                        bfs_strsplit(path, "/", i));
-                    walker->child = node;
-                    node->father = walker;
-                }
+                printf("%s   ", root->fname);
             }
-
-
-            i++;
+            
+            root = root->next;
         }
+        newline;
+        return;
     }
+    else if(strcmp(arg, "-i") == 0) {
+        newline;
+        printf("Type\tName\t\t\tCreated on\n\n");
+
+        while(root != NULL) {
+            if(root->fpermissions == BFS_NODE_IS_DIR) {
+                printf("-d-\t%s%-20s%s\t%s", 
+                    ANSI_COLOR_BLUE, root->fname, ANSI_COLOR_RESET, 
+                    asctime(gmtime(&(root->fcreated))));
+            }
+            else {
+                printf("-f-\t%-20s\t%s", root->fname, asctime(gmtime(&(root->fcreated))));
+            }
 
-
-    // FIXME: Complete this thing. Walker is undefined outside the second if
-    /*while (bfs_strsplit(path, "/", 1) != walker->fname &&
-           walker->next != NULL) {
-        walker = walker->next;
+            root = root->next;
+        }
+        return;
     }
-
-    if(walker->next == NULL) {
-
-    }*/
-}
-
-int bfs_tree_insert(BFSTree *root, char path[BFS_PATH_SIZE]) {
-    if(root == NULL) {
-        return 0;
+    else if(strcmp(arg, "-h") == 0) {
+        goto help;
```

Comment:
Although I don't usually prohibit this kind of thing, this in this case is not any useful. Can you please create a function to show specific help inside BFS?

Reply:
Fixed.